### ISSUE TITLE
Nonce getting regenerated and rejected.

### ISSUE THREAD

**Issue by leticiatm** [WP-API/WP-API#2676]
I'm having a weird situation. 
Having two users, the one that was created when first setting up WP, and a second user I created using wp_create_user(), even if both are set as admin, only the first gets permission to create a new post. The second gets a 403 error with the message "Cookie nonce is invalid". No custom endpoint, just using w2/v2/cupom which cupom is a custom post type only.

So I started debugging the nonce code, it's working fine from php to the js, it's sending the ajax request correctly, but when I reloaded the page to retry, the nonce was regenerated, while the first user gets a 200 code and the nonce is the same as before the request.

What is happing? Is wp_create_user not working fine with nonce? What can I do to make it work?

The request:

```
jQuery.ajax({
                dataType: 'json',
                url: meepondata.couponurl,
                method: 'POST',
                beforeSend: function (xhr) {
                    xhr.setRequestHeader('X-WP-Nonce',meepondata.wprest_api.nonce);

                },
                data: {
                    'title': codigo,
                    'slug': codigo,
                    'fields[promocao]': idpromocao,
                    'fields[usuario]': usuario,
                    'fields[codigo]': codigo,
                    'fields[empresa]': idresponsavel,
                    'status': 'publish'
                }

            })
            .done(function (response) {
console.log(response);
                window.location = "http://localhost/meepon/cupom/" + codigo;

            });
```

UPDATE: found out that the issue happens after I register/login the user using ajax/wp api, because if the user is already logged in, it works fine. But I'm sending the new nonce after all the process, so I don't get what's going on. This is the function that it's called to check user info and log the person in.

```
function user_cpf( $data ) {
    if(is_user_logged_in()){
        $res = array('message'=>"islogged",'result'=>'error');
        return $res;
    }
    $cpf = $data['cpf'];
    $email = $data['email'];

    $user = get_user_by_meta_data( 'cpf', $cpf );

    if ( empty( $user->ID ) ) {
        //registers and login
        if(email_exists($email) === false){
            $user_id = wp_create_user( $cpf, $cpf, $email );
            add_user_meta( $user_id, 'cpf', $cpf);

            $res = array('result'=>'success','new'=>true,'idUser'=>$user_id);

        }
        else{
            $res = array('message'=>"Email já cadastrado com outro CPF",'result'=>'error','issue'=>'email');
        }
    }
    else{
        //user exists, but let's check if the email is the same as registered
        if($email === $user->user_email){
            $res = array('result'=>'success','new'=>false,'idUser'=>$user->ID);
        }else{
            $res = array('message'=>"CPF já cadastrado com outro email",'result'=>'error','issue'=>'cpf');
        }
    }
    if($res['result'] === 'success'){

        $creds = array();
        $creds['user_login'] = $email;
        $creds['user_password'] = $cpf;
        $creds['remember'] = true;
        $user = wp_signon( $creds, false );
        if (! is_wp_error($user) ){
            wp_set_current_user($user->ID);

        }
        $res['newNonce'] =  wp_create_nonce( 'wp_rest' );

    }
    return $res;


}
```

So now there're three different nonce codes: the one when the user is logged out (before the process), the one I'm sending back after logging the user in, and the last one, that I think it's the valid one but I don't see how to get it, appears when I refresh the page and retest the nonce code (echoing it )

**Comment by BE-Webdesign** [comment]
Hi @leticiatm,

Thank you for using the WP REST API! Can you try to explain what you are trying to accomplish first.

Try logging in as both users separately and creating the cupom post via the API, it will probably work. Switching users within the PHP execution will invalidate your cookies and the wp-api.js authentication is not meant to be used in the way that I believe you are trying to use it (I'm still not 100% sure if I follow what you are trying to do).

**Comment by leticiatm** [comment]
I already tried, same result. I'm actually using incognito mode, to make sure all cookies and sessions are clean.
I'm trying to log in/register the user, then create a new coupon (cupom), all without refreshing the page, using two ajax calls for that. 
If the user is already logged in, it doesn't run that second function, user_cpf, so it works beautifully. But if I need to log in/register the person, even sending the nonce AFTER the user is logged, it says it's wrong. 
It looks like there are three nonces involved: the one when it's not logged, the one after it's logged, and I'm passing to JS ( $res['newNonce'] =  wp_create_nonce( 'wp_rest' ); ), and the one that looks like it's the right, that shows up only after I refresh the page (I'm echoing it so I can test it), but I'm not seeing a way to get it without refreshing the page.

Check if logged(JS) -> if not, ajax call to register/log in (JS->PHP) -> Log in and send new nonce back (PHP->JS) - Ajax call to create coupon (JS->PHP)

I'm thinking of giving up on this idea and create the coupon right after the log in, in the same ajax, so there's no error.

**Comment by nylen** [comment]
Now that the REST API has been merged into WordPress core, development and support are no longer taking place in this repository.

If you're still having this issue, please post a support request to the WordPress forums at https://wordpress.org/support/forum/how-to-and-troubleshooting/ with the topic tag "rest-api".

If you're able to identify clear steps to reproduce the bug from a clean install of WordPress 4.7, please post a Trac ticket at https://core.trac.wordpress.org -- Be sure to include full details and reproduction steps about the issue you are experiencing.

If you are unfamiliar with Trac, you can refer to the "Opening a Ticket" user guide at https://make.wordpress.org/core/handbook/tutorials/trac/opening-a-ticket/ for an introduction!
